TestFailedReplicaChange (and others) appear to have leaked an rpc client

From the deployment section in https://circleci.com/gh/cockroachdb/cockroach/1089:
```
--- FAIL: TestFailedReplicaChange (0.06s)
	leaktest.go:135: Test appears to have leaked an rpc client:
		net.(*pollDesc).Wait(0xc2084a8060, 0x72, 0x0, 0x0)
			/usr/src/go/src/net/fd_poll_runtime.go:84 +0x47
		net.(*pollDesc).WaitRead(0xc2084a8060, 0x0, 0x0)
			/usr/src/go/src/net/fd_poll_runtime.go:89 +0x43
		net.(*netFD).Read(0xc2084a8000, 0xc208433000, 0x1000, 0x1000, 0x0, 0x7fb9ac060e20, 0xc20821ca98)
			/usr/src/go/src/net/fd_unix.go:242 +0x40f
		net.(*conn).Read(0xc20821e000, 0xc208433000, 0x1000, 0x1000, 0x0, 0x0, 0x0)
			/usr/src/go/src/net/net.go:121 +0xdc
		bufio.(*Reader).fill(0xc20843a000)
			/usr/src/go/src/bufio/bufio.go:97 +0x1ce
		bufio.(*Reader).Read(0xc20843a000, 0xc20821c030, 0x1, 0x9, 0xc208151940, 0x0, 0x0)
			/usr/src/go/src/bufio/bufio.go:174 +0x26c
		io.ReadAtLeast(0x7fb9ac064738, 0xc20843a000, 0xc20821c030, 0x1, 0x9, 0x1, 0x0, 0x0, 0x0)
			/usr/src/go/src/io/io.go:298 +0xf1
		io.ReadFull(0x7fb9ac064738, 0xc20843a000, 0xc20821c030, 0x1, 0x9, 0x1, 0x0, 0x0)
			/usr/src/go/src/io/io.go:316 +0x6d
		encoding/gob.decodeUintReader(0x7fb9ac064738, 0xc20843a000, 0xc20821c030, 0x9, 0x9, 0x0, 0x1, 0x0, 0x0)
			/usr/src/go/src/encoding/gob/decode.go:121 +0x99
		encoding/gob.(*Decoder).recvMessage(0xc2080ee100, 0xc2081b1440)
			/usr/src/go/src/encoding/gob/decoder.go:76 +0x55
		encoding/gob.(*Decoder).decodeTypeSequence(0xc2080ee100, 0xc20813f700, 0x16)
			/usr/src/go/src/encoding/gob/decoder.go:140 +0x47
		encoding/gob.(*Decoder).DecodeValue(0xc2080ee100, 0xc99880, 0xc20813f760, 0x16, 0x0, 0x0)
			/usr/src/go/src/encoding/gob/decoder.go:208 +0x192
		encoding/gob.(*Decoder).Decode(0xc2080ee100, 0xc99880, 0xc20813f760, 0x0, 0x0)
			/usr/src/go/src/encoding/gob/decoder.go:185 +0x297
		net/rpc.(*gobServerCodec).ReadRequestHeader(0xc208138180, 0xc20813f760, 0x0, 0x0)
			/usr/src/go/src/net/rpc/server.go:402 +0x5e
		net/rpc.(*Server).readRequestHeader(0xc2083d8c80, 0x7fb9ac072f60, 0xc208138180, 0x0, 0x0, 0xc20813f760, 0xc20821c000, 0x0, 0x0)
			/usr/src/go/src/net/rpc/server.go:575 +0x97
		net/rpc.(*Server).readRequest(0xc2083d8c80, 0x7fb9ac072f60, 0xc208138180, 0xc2080ee600, 0xc20813f700, 0xc901c0, 0x0, 0x0, 0x0, 0x0, ...)
			/usr/src/go/src/net/rpc/server.go:542 +0xa2
		net/rpc.(*Server).ServeCodec(0xc2083d8c80, 0x7fb9ac072f60, 0xc208138180)
			/usr/src/go/src/net/rpc/server.go:461 +0x6d
		net/rpc.(*Server).ServeConn(0xc2083d8c80, 0x7fb9ac072e48, 0xc20821e000)
			/usr/src/go/src/net/rpc/server.go:453 +0x208
		github.com/cockroachdb/cockroach/multiraft.func·007(0x7fb9ac072de8, 0xc20821e000)
			/go/src/github.com/cockroachdb/cockroach/multiraft/transport.go:136 +0xda
		created by github.com/cockroachdb/cockroach/multiraft.(*localRPCTransport).accept
			/go/src/github.com/cockroachdb/cockroach/multiraft/transport.go:137 +0x3c7
		
		net.(*pollDesc).Wait(0xc2084a8450, 0x72, 0x0, 0x0)
			/usr/src/go/src/net/fd_poll_runtime.go:84 +0x47
		net.(*pollDesc).WaitRead(0xc2084a8450, 0x0, 0x0)
			/usr/src/go/src/net/fd_poll_runtime.go:89 +0x43
		net.(*netFD).Read(0xc2084a83f0, 0xc208427000, 0x1000, 0x1000, 0x0, 0x7fb9ac060e20, 0xc20821cec8)
			/usr/src/go/src/net/fd_unix.go:242 +0x40f
		net.(*conn).Read(0xc20821ec08, 0xc208427000, 0x1000, 0x1000, 0x0, 0x0, 0x0)
			/usr/src/go/src/net/net.go:121 +0xdc
		bufio.(*Reader).fill(0xc20843b080)
			/usr/src/go/src/bufio/bufio.go:97 +0x1ce
		bufio.(*Reader).Read(0xc20843b080, 0xc20821db30, 0x1, 0x9, 0xc2080ef850, 0x0, 0x0)
			/usr/src/go/src/bufio/bufio.go:174 +0x26c
		io.ReadAtLeast(0x7fb9ac064738, 0xc20843b080, 0xc20821db30, 0x1, 0x9, 0x1, 0x0, 0x0, 0x0)
			/usr/src/go/src/io/io.go:298 +0xf1
		io.ReadFull(0x7fb9ac064738, 0xc20843b080, 0xc20821db30, 0x1, 0x9, 0xa0eb0d, 0x0, 0x0)
			/usr/src/go/src/io/io.go:316 +0x6d
		encoding/gob.decodeUintReader(0x7fb9ac064738, 0xc20843b080, 0xc20821db30, 0x9, 0x9, 0x0, 0x1, 0x0, 0x0)
			/usr/src/go/src/encoding/gob/decode.go:121 +0x99
		encoding/gob.(*Decoder).recvMessage(0xc2080ef800, 0x1)
			/usr/src/go/src/encoding/gob/decoder.go:76 +0x55
		encoding/gob.(*Decoder).decodeTypeSequence(0xc2080ef800, 0xc208138900, 0x16)
			/usr/src/go/src/encoding/gob/decoder.go:140 +0x47
		encoding/gob.(*Decoder).DecodeValue(0xc2080ef800, 0xc998e0, 0xc2081389f0, 0x16, 0x0, 0x0)
			/usr/src/go/src/encoding/gob/decoder.go:208 +0x192
		encoding/gob.(*Decoder).Decode(0xc2080ef800, 0xc998e0, 0xc2081389f0, 0x0, 0x0)
			/usr/src/go/src/encoding/gob/decoder.go:185 +0x297
		net/rpc.(*gobClientCodec).ReadResponseHeader(0xc208138900, 0xc2081389f0, 0x0, 0x0)
			/usr/src/go/src/net/rpc/client.go:223 +0x5e
		net/rpc.(*Client).input(0xc20843b140)
			/usr/src/go/src/net/rpc/client.go:109 +0xb6
		created by net/rpc.NewClientWithCodec
			/usr/src/go/src/net/rpc/client.go:201 +0xd0
		
		testing.RunTests(0xfed2a0, 0x1465080, 0x89, 0x89, 0xc207fff301)
			/usr/src/go/src/testing/testing.go:556 +0xad6
		testing.(*M).Run(0xc20800cb90, 0x50)
			/usr/src/go/src/testing/testing.go:485 +0x6c
		github.com/cockroachdb/cockroach/util/leaktest.TestMainWithLeakCheck(0xc20800cb90)
			/go/src/github.com/cockroachdb/cockroach/util/leaktest/leaktest.go:34 +0x28
		github.com/cockroachdb/cockroach/storage.TestMain(0xc20800cb90)
			/go/src/github.com/cockroachdb/cockroach/storage/main_test.go:29 +0x28
		main.main()
			github.com/cockroachdb/cockroach/storage/_test/_testmain.go:324 +0x1d1
```

--- a/storage/client_raft_test.go
+++ b/storage/client_raft_test.go
@@ -274,6 +274,23 @@ func TestFailedReplicaChange(t *testing.T) {
 	if err != nil {
 		t.Fatal(err)
 	}
+
+	// Wait for the range to sync to both replicas (mainly so leaktest doesn't
+	// complain about goroutines involved in the process).
+	if err := util.IsTrueWithin(func() bool {
+		for _, store := range mtc.stores {
+			rng, err := store.GetRange(1)
+			if err != nil {
+				return false
+			}
+			if len(rng.Desc().Replicas) == 1 {
+				return false
+			}
+		}
+		return true
+	}, 1*time.Second); err != nil {
+		t.Fatal(err)
+	}
 }
 
 // We can truncate the old log entries and a new replica will be brought up from a snapshot.

